File: cut-vector.py
Here is the completed code:

def cut_vector(nums):
    min_diff = float('inf')
    cut_idx = -1
    
    for i in range(1, len(nums)):
        left_sum = sum(nums[:i])
        right_sum = sum(nums[i:])
        
        diff = abs(left_sum - right_sum)
        if diff < min_diff:
            min_diff = diff
            cut_idx = i
            
    return nums[:cut_idx], nums[cut_idx:]

File: cut-vector.cpp
#include <iostream>
#include <vector>
#include <numeric>
#include <climits>
#include <cmath>

using namespace std;

int main() {
    vector<int> nums;
    int num;
    
    int diff = INT_MAX;
    int idx = -1;
    
    while (!cin.eof() && cin >> num) {
        nums.push_back(num);

        int len = nums.size();
        int left_sum = accumulate(nums.begin(), nums.begin() + len, 0);
        int right_sum = accumulate(nums.begin() + len, nums.end(), 0);

        if (abs(left_sum - right_sum) < diff) {
            diff = abs(left_sum - right_sum);
            idx = len;
        }
    }
    
    for (int i = 0; i < idx; ++i) {
        cout << nums[i] << endl;
    }
    
    cout << endl;
    
    for (int i = idx; i < nums.size(); ++i) {
        cout << nums[i] << endl;
    }
    
    return 0;
}